src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs

```diff
@@ -238,6 +238,7 @@ static void F(System.Collections.Generic.IEnumerable<(int a, int b)> ie)
           <slot kind=""0"" offset=""59"" />
           <slot kind=""0"" offset=""62"" />
           <slot kind=""temp"" />
+          <slot kind=""temp"" />
```

Comment:
In a d-assignment into two variables there were always two additional temps (that receive the outputs of conversion stage and are the inputs for assignment stage).
In a d-declaration, previous, one of those two temps was optimized away (because a deconstruction declaration was always returning void). Now, a d-declaration (and d-foreach) prepare bound nodes to construct a result. Constructing this ValueTuple is optimized away, but the optimizer is apparently not smart enough to remove the temp as before.

I'm wondering if the d-foreach lowering should produce bound nodes for constructing a ValueTuple...

---
In reply to: [88575521](https://github.com/dotnet/roslyn/pull/15192#discussion_r88575521) [](ancestors = 88575521)